modules/examples/pages/defi.adoc

```diff
@@ -0,0 +1,81 @@
+```candid
+placeOrder: (Token, nat, Token, nat) -> (OrderPlacementReceipt);
+```
+
+Allows the user to cancel submitted orders.
+```candid
+cancelOrder: (OrderId) -> (CancelOrderReceipt);
+```
+
+Request user's balance on exchange for a specific token.
+```candid
+getBalance: (Token) -> (nat) query;
+```
+
+### Fee
+
+It is the responsibility of the exchange to subtract fees from the trades. This is important because the exchange must pay fees for withdraws and internal transfers.
+
+## Token Exchange Walkthrough
+This section contains a detailed walkthrough of the core exchange functionalities. Most interactions require multiple steps and are simplified by using the provided frontend. Since the exchange canister functions are public, advanced users can use ``dfx`` to interact with the exchange.
+
+### Depositing ICP
+
+The ledger canister provides a unique interface and therefore, interactions with ICP need to be resolved separately.
+
+* The user calls the ``getDepositAddress`` function. The response contains a unique account identifier representing a user-specific subaccount controlled by the exchange. The exchange can identify the user responsible for deposits through this address.
+* User transfers ICP to the fetched deposit address and waits for the transfer to complete.
+* To notify the exchange, the user calls ``deposit`` with the ICP token principal. The exchange will look into the user's subaccount and adjust the user's balance on the exchange. In a second step, the exchange will transfer the funds from the user subaccount to its default subaccount, where the exchange keeps all of its ICP.
+
+### Depositing Tokens
+
+Depositing tokens is more straightforward because DIP20 provides a richer interface to interact with.
+
+* The user calls the ``approve`` function of the token canister. This gives the exchange the ability to transfer funds to itself on behalf of the user.
+* Similar to the ICP depositing, the user calls the ``deposit`` function of the exchange. The exchange then transfers the approved token funds to itself and adjusts the user's exchange balance.
+
+### Placing Orders
+
+After depositing funds to the exchange, the user can place orders. An order consists of two tuples. ``from: (Token1, amount1)`` and ``to: (Token2, amount2)``. These orders get added to the exchange. What happens to these orders is specific to the exchange implementation. This example provides a simple exchange that only executes exactly matching orders. Be aware this is just a toy exchange, and the exchange functionality is just for completeness. Hint: The exchange can be greedy sometimes ;)
+
+### Withdrawing Funds
+
+Compared to depositing funds, withdrawing funds is simpler. Since the exchange has custody of the funds, the exchange will send funds back to the user on ``withdraw`` requests. The internal exchange balances are adjusted accordingly.
+
+
+## Common mistakes
```

Comment:
Are these address in the current implementation or lingering bugs?

Reply:
These are more of "gotchas" to look out for while developing, that we had run into.